src/Loop.php

```diff
@@ -59,155 +59,162 @@ public static function stop()
     /**
      * Defer the execution of a callback.
      *
-     * @param callable $callback The callback to defer.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to defer.
```

Comment:
Here (defer) and for delay(), why do we pass a $watcherIdentifier? It's going to be immediately invalid as soon as the callback is left. The $watcherIdentifier is not meaningful in any case. If you need to map some data to something or such, that's what the $data parameter is for.